### DISCUSSION TITLE
Copy a set of files/trees from one repo to another repo via GitHub API?

### DISCUSSION THREAD

**Discussion by chenleshang** [community/community#24891] [Apps, API and Webhooks]
<p>Hi! Is there any way to copy a set of files/trees from one repo to another repo via GitHub API? Not by reading the content of the file from one repo and reupload to another repo, but some other ways e.g. through the tree SHA / blob SHA in the original repo? </p>

**Comment by lee-dohm** [comment, marked as answer]
<p>In theory, yes. In practice, you probably don’t want to <a href="https://www.hanselman.com/blog/YakShavingDefinedIllGetThatDoneAsSoonAsIShaveThisYak.aspx" rel="nofollow noopener">shave that particular yak</a>. Since there isn’t a magical “take this from here and put it there” API, you would still have to:</p>
<ol>
<li>
<a href="https://developer.github.com/v3/git/blobs/#get-a-blob" rel="nofollow noopener">Download all the appropriate blob data</a> from the source repo</li>
<li>Upload all the blob data to the destination repo by <a href="https://developer.github.com/v3/git/blobs/#create-a-blob" rel="nofollow noopener">creating individual blob objects</a>
</li>
<li>
<a href="https://developer.github.com/v3/git/trees/#create-a-tree" rel="nofollow noopener">Create the recursive forest of tree objects</a> necessary to reference all the blobs you just uploaded</li>
<li>Then <a href="https://developer.github.com/v3/git/commits/#create-a-commit" rel="nofollow noopener">create a commit object</a> to root the trees you just planted on the appropriate branch</li>
</ol>
<p>Or, you could have Git do it because it knows how to manipulate its own database.</p>
